RoboticonColony/Assets/Code/Market.cs

```diff
@@ -129,4 +138,89 @@ public Market Sell(ItemType item, int quantity, Inventory playerInventory)
             throw;
         }
     }
+ 
+    /// <summary>
+    /// Creates an customisation type for roboticons.
+    /// </summary>
+    /// <param name="selectedName"> The name of the customisation. </param>
+    /// <param name="bonusMult"> The multiplier which the production will be boosted by. </param>
+    /// <param name="prereq"> The list of other customisations which must completed already before this customisation can be applied. </param>
+    /// <param name="selectedResource"> The type of resource which the customisation augments. </param>
+    /// <param name="reqPrice"> The required price of the customisation. </param>
+    private void CreateCustomisations(string selectedName, Dictionary<ItemType, int> bonuses, List<RoboticonCustomisation> prereq, ItemType selectedResource, int reqPrice)
+    {
+        CustomisationsList.Add(new RoboticonCustomisation(selectedName, bonuses, prereq, reqPrice));
+    }
+
+    public int GetTilePrice(Tile tile)
+    {
+        return tile.Price;
+    }
+  
+    /// <summary>
+    /// Adds a roboticon to the market stock.
+    /// </summary>
+    public void BuyRoboticonOre()
+    {
+        //TODO: Update to not use negative numbers, as this will no longer work with inventory in future
+        if (Stock.GetItemAmount(ItemType.Ore) > 0)
+        {
+            Stock.AddItem(ItemType.Ore, -1);
+            Stock.AddItem(ItemType.Roboticon, 1);
+        }
+    }
+
+    /// <summary>
+    /// Function to buy a specified customisation for the specified roboticon.
+    /// </summary>
+    /// <param name="inv"> The player's inventory</param>
+    /// <param name="customisation"> The selected roboticon customisation</param>
+    /// <param name="roboticon"> The selected Roboticon</param>
+    public void BuyCustomisation(RoboticonCustomisation customisation, Roboticon roboticon, Inventory inventory)
+    {
+        if (inventory.Money > customisation.Price)
+        {
+            roboticon.AddCustomisation(customisation);
+            inventory.AddMoney(-customisation.Price);
```

Comment:
I think we will have to add a SubtractMoney method in inventory for future use, it just makes it look nicer